lib/lan_connection.js

```diff
@@ -38,80 +40,71 @@ LANConnection.prototype.end = function() {
+    // Open up an SSH Connection
+    self.ssh.on('ready', function() {
+      // Tessel allows connection
+      self.authorized = true;
+      // Resolve with connection
+      resolve(self);
+    });
+    self.ssh.on('error', function(err) {
+      // Reject with error
+      reject(err);
+    }).connect(self.auth);
   });
-  this.ssh.on('error', function(err) {
-    callback && callback(err);
-  }).connect(this.auth);
 }
 
 // Find all LAN connected Tessels
-function findConnections(callback) {
-
+function findConnections() {
+  var lanConfig, envConfig;
   // Grab all mDNS connected Tessels
-  findmDNSConnections(function(err, connections) {
-    // Riase any errors
-    if (err) {
-      return callback && callback(err);
-    }
-
-    // Check if a config file exists
-    fs.exists(__dirname + '/../config.env', function (exists) {
-      // If it does
-      if (exists) {
-        // parse it
-        config = envfile.parseFileSync(__dirname + '/../config.env')
-        // Create a LAN Connection with the provided auth credentials
-        var conn = new LANConnection({
-          host: config.host,
-          username: 'root',
-          port: 22,
-          privateKey: fs.readFileSync(config.keyPath),
-          passphrase: config.keyPassphrase
+  return findmDNSConnections()
+    // TODO: Remove this step once Tessel Setup is built
+    .then(function(tessels){
+      return new Promise(function(resolve, reject) {
+        // Check if a config file exists
+        fs.exists(__dirname + '/../config.env', function (exists) {
+          // If it does
+          if (exists) {
+            // parse it
+            envConfig = envfile.parseFileSync(__dirname + '/../config.env')
+            // Create a LAN Config with the provided auth credentials
+            lanConfig = {
+              host: envConfig.host,
+              username: 'root',
+              port: 22,
+              privateKey: fs.readFileSync(envConfig.keyPath),
+              passphrase: envConfig.keyPassphrase
+            }
+          } else {
+            return reject('Set up your config.env with keyPath and keyPassphrase');
+          }
+          // Pass the tessel's through or else you have no tessels
+          return resolve(tessels);
         });
-
-        // Insert the specified connection as the first item
-        // so that it will be preferred
-        connections.unshift(conn);
-      }
-
-      // Open the connections and return them
-      return openConnections(connections, callback);
-    });
-  });
-
-  // For each LAN Conenction, try to open up an SSH session
-  // Should we really be doing this here?
-  function openConnections(connections, callback) {
-    async.each(connections, function(conn, cb) {
-      conn.open(function(err) {
-        conn.initError = err;
-        cb(err);
       });
-    }, function(err) {
-      callback(err, connections)
-    });
-  }
+    })
+    // Create new LAN connections for each tessel.
+    .map(function(tessel){
```

Comment:
Previously, the "preferred" Tessel was the first Tessel in the returned Tessel list. Admittedly, there is a better way to handle that, but for now, this logic places the Tessel selected by the `config.env` somewhere randomly in the list. It needs to be first.